wasm/bindings.cpp

```diff
@@ -0,0 +1,113 @@
+#include <emscripten/bind.h>
+
+using namespace emscripten;
+
+#include <manifold.h>
+
+using namespace manifold;
+
+void logMesh(Mesh& mesh) {
+    printf("# %d vertices:\n", (int)mesh.vertPos.size());
+    for(int i = 0; i < mesh.vertPos.size(); i++) {
+        printf("v %f %f %f\n",
+            mesh.vertPos[i].x,
+            mesh.vertPos[i].y,
+            mesh.vertPos[i].z
+        );
+    }
+    printf("# %d faces:\n", (int)mesh.triVerts.size());
+    for(int i = 0; i < mesh.triVerts.size(); i++) {
+        printf("f %d %d %d\n",
+            mesh.triVerts[i].x + 1,
+            mesh.triVerts[i].y + 1,
+            mesh.triVerts[i].z + 1
+        );
+    }
+}
+
+struct Mesh makeEmptyMesh() {
+  struct Mesh empty; return empty;
+}
+
+Manifold makeManifoldFromMesh(Mesh& mesh) {
+    //logMesh(mesh);
+
+    try {
+        Manifold manifold(mesh);
+        return manifold;
+    }
+    catch(const std::runtime_error& e) {
+        puts(e.what());
+
+        Manifold empty;
+        return empty;
+    }
+}
+
+Mesh makeMeshFromManifold(Manifold& manifold) {
+    try {
+        Mesh mesh = manifold.GetMesh();
+        //logMesh(mesh);
+
+        return mesh;
+    }
+    catch(const std::runtime_error& e) {
+        puts(e.what());
+
+        Mesh empty;
+        return empty;
+    }
+}
+
+void subtract(Manifold& a, Manifold& b) {
+    a -= b;
+}
+
+EMSCRIPTEN_BINDINGS(whatever) {
```

Comment:
Love it; does this name even show up anywhere in the actual WASM blob?

Reply:
it is not in manifold.js at least; emscripten docs [say](https://emscripten.org/docs/api_reference/bind.h.html#_CPPv419EMSCRIPTEN_BINDINGS4name) this:
> a label to mark a group of related bindings (for example EMSCRIPTEN_BINDINGS(physics), EMSCRIPTEN_BINDINGS(components), etc.)